components/core/Settings/Jumper.js

```diff
@@ -0,0 +1,412 @@
+  };
+
+  const _handleUpload = async (e) => {
+    setState((prev) => ({ ...prev, changingAvatar: true }));
+    
+    let file = await UserBehaviors.uploadImage(e.target.files[0]);
+    if (!file) {
+      setState((prev) => ({ ...prev, changingAvatar: false }));
+      return;
+    }
+
+    const cid = file.cid;
+    const url = Strings.getURLfromCID(cid);
+    
+    let updateResponse = await Actions.updateViewer({
+      user: {
+        photo: url,
+      },
+    });
+
+    Events.hasError(updateResponse);
+    setState((prev) => ({ ...prev, changingAvatar: false, photo: url }));
+  };
+
+  const _handleSave = async (e) => {
+    if (!Validations.username(state.username)) {
+      Events.dispatchMessage({
+        message: "Please include only letters and numbers in your username",
+      });
+      return;
+    }
+    /*
+    props.onAction({
+      type: "UPDATE_VIEWER",
+      viewer: { username: state.username, name: state.name, body: state.body },
+    });
+    */
+    setState((prev) => ({ ...prev, savingNameBio: true }));
+
+    let response = await Actions.updateViewer({
+      user: {
+        name: state.name,
+        photo: state.photo,
+        body: state.body,
+      },
+    });
+
+    Events.hasError(response);
+    setState((prev) => ({ ...prev, savingNameBio: false }));
+  };
+
+  return (
+    <>
+      <div css={STYLES_JUMPER_CONTENT}>
+        <div css={STYLES_AVATAR_BOX}>
+          <p css={STYLES_ITEM_HEADER}>Avatar</p>
+          <div css={STYLES_AVATAR_UPLOAD}>
+            <ProfilePhoto
+              user={state}
+              size={48}
+              style={{ borderRadius: '10px' }}
+            />
+
+            <input 
+              type="file"
+              id="file-input-id"
+              css={STYLES_FILE_HIDDEN}
+              onChange={_handleUpload}
+            />
+
+            <ButtonSecondary
+              style={{ marginLeft: "8px", minHeight: "24px", borderRadius: '8px' }}
+              type="label"
+              htmlFor="file-input-id"
+              loading={state.changingAvatar}
+            >
+              Upload
+            </ButtonSecondary>
+
+          </div>
+        </div>
+
+        <div css={STYLES_ITEM_CONTAINER}>
+          <p css={STYLES_ITEM_HEADER}>Display name</p>
+          <Input
+            name="name"
+            value={state.name}
+            placeholder="Your name..."
+            maxLength="255"
+            onChange={_handleChange}
+          />
+        </div>
+
+        <div css={STYLES_ITEM_CONTAINER}>
+          <p css={STYLES_ITEM_HEADER}>Bio</p>
+          <Textarea
+            name="body"
+            value={state.body}
+            placeholder="A bit about yourself..."
+            maxLength="2000"
+            onChange={_handleChange}
+          />
+        </div>
+
+        <div css={STYLES_EDIT_INFO_FOOTER}>
+          <ButtonSecondary 
+            style={{ minHeight: "24px", marginLeft: "auto", borderRadius: '8px' }}
```

Comment:
you would need to add paddings to all secondary buttons in smaller size.
refer to design for the numbers.